src/sinks/hdfs/mod.rs

```diff
@@ -0,0 +1,75 @@
+//! `hdfs` sink.
+//!
+//! This sink will send it's output to hdfs.
+//!
+//! `hdfs` is an opendal based services. This mod itself only provide
+//! config to build an [`OpendalSink`]. All real implement are powered by
+//! [`OpendalSink`].
+
+use super::opendal_common::*;
+use super::Healthcheck;
+use crate::config::{GenerateConfig, SinkConfig, SinkContext};
+use opendal::services::Hdfs;
+use opendal::Operator;
+use vector_config::configurable_component;
+use vector_core::{
+    config::{AcknowledgementsConfig, Input},
+    sink::VectorSink,
+};
+
+/// A sink that dumps its output to hdfs.
+#[configurable_component(sink("hdfs"))]
+#[derive(Clone, Debug)]
+pub struct HdfsConfig {
+    /// A prefix/root to apply to all pathes.
+    #[serde(default)]
+    #[configurable(metadata(docs::templateable))]
+    pub root: String,
+    #[configurable(derived)]
+    #[serde(default)]
+    pub name_node: String,
+    #[configurable(derived)]
+    #[serde(
+        default,
+        deserialize_with = "crate::serde::bool_or_struct",
+        skip_serializing_if = "crate::serde::skip_serializing_if_default"
+    )]
+    pub acknowledgements: AcknowledgementsConfig,
```

Comment:
```suggestion
    /// A prefix/root to apply to all pathes.
    #[serde(default)]
    #[configurable(metadata(docs::templateable))]
    pub root: String,
    
    /// TODO user facing description of a NameNode, possibly link to HDFS docs
    #[configurable(derived)]
    #[serde(default)]
    pub name_node: String,
    
    #[configurable(derived)]
    #[serde(
        default,
        deserialize_with = "crate::serde::bool_or_struct",
        skip_serializing_if = "crate::serde::skip_serializing_if_default"
    )]
    pub acknowledgements: AcknowledgementsConfig,
```

**chore:** Add spacing for better readability.

We'd also want some more descriptive and user friendly comments for the two options (with links if possible).